steely/plugins/obscene.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+"""
+Detects when people say naughty things and punishes them with a badge.
+"""
+
+import requests
+from string import punctuation
+
+from plugins import gex
+
+__author__ = 'CianLR'
+COMMAND = None
+BAD_WORDS_URL = ('https://raw.githubusercontent.com/LDNOOBW/'
+                 'List-of-Dirty-Naughty-Obscene-and-Otherwise-Bad-Words/'
+                 'master/en')
+BAD_WORDS = set(requests.get(BAD_WORDS_URL).text.split('\n'))
+
+card_made = False
+CIAN_ID = '1845973042'
+CARD_NAME = "obscene"
+CARD_DESC = "WARNING: This person is not fit for a christian minecraft server"
+CARD_IMAGE = "https://pics.me.me/this-a-christian-page-swearing-21946873.png"
+
+def main(bot, author_id, message, thread_id, thread_type, **kwargs):
+    global card_made
+    if not card_made:
+        gex.gex_create(CARD_NAME, [bot.uid, CIAN_ID], CARD_DESC)
```

Comment:
this will throw a runtimeerror or something on subsequent steely boots after the first one, because the card will already exist

Reply:
Ah, assumed it was idempotent. I've changed this to search for it in the codex on import. 

We're really going to need a nice API for pre-registering cards :stuck_out_tongue:

Reply:
if you'd pop it in a try-except it's totally a-ok and idempotent. Thought it was more useful to give an error than just to do nothing.

Reply:
Hm yeah. Too much C++ has given me a wariness of exceptions. Although they're a lot safer in Python.
I guess there's nothing wrong with `gex_create` erroring, the best fix is probably not calling it in the first place if you don't need to